File: scripts/navmenu.py
# -*- coding: utf-8 -*-
import re, os
import os, sys
# roda a partir da raiz do repositório, não importa de onde foi chamado
os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))

# (href, label_forte, label_descricao)
GRUPOS = [
    ('Construção', [
        ('html-puro.html', 'HTML puro', 'Do zero ao site publicado'),
        ('qualidade.html', 'Qualidade', 'Imagem, a11y, Vitals e testes'),
        ('movimento.html', 'Movimento', 'Animação com critério'),
        ('motion.html', 'Referência: Motion', 'A API inteira, em português'),
        ('frameworks.html', 'Frameworks', 'Formatos além do vanilla'),
        ('back-end.html', 'Back-end', 'Servidor, banco e login'),
    ], None),
    ('Captação e Vendas', [
        ('landing-pages.html', 'Landing pages', 'A estratégia da página'),
        ('trafego.html', 'Tráfego', 'Ser encontrado no Google'),
        ('negocio.html', 'Negócio', 'Preço, escopo e contrato'),
    ], None),
    ('Método', [
        ('fluxo.html', 'Fluxo de trabalho', 'Terminal, Git e DevTools'),
        ('ia.html', 'Trabalhando com IA', 'Brief, referência e o gerador'),
        ('colofao.html', 'Colofão', 'Como este site foi feito'),
    ], None),
    ('Ferramentas', [
        ('ferramentas.html', 'Ferramentas', 'As doze do dia a dia'),
        ('ferramentas.html#f-checklist', 'Checklist de entrega', 'Antes de publicar'),
        ('ferramentas.html#f-proposta', 'Proposta que fecha', 'Os oito pontos'),
        ('ferramentas.html#f-briefing', 'Briefing', 'As dez perguntas'),
        ('ferramentas.html#f-zap', 'Link de WhatsApp', 'Com a mensagem pronta'),
        ('ferramentas.html#f-utm', 'Gerador de UTM', 'Rastrear a origem'),
        ('ia.html#t-ia-gerador', 'Gerador de comando', 'Brief para a IA'),
    ], 'navmenu__grupo--trabalhar'),
    ('Patch notes', [
        ('patch-notes.html', 'Patch notes', 'O que mudou no site'),
    ], None),
]

# arquivo -> (href que fica 'is-atual', texto do navmenu__aqui) — None = nenhum dos dois (home/404)
# arquivo -> modo do alternador ('estudo', 'trabalho' ou None)
MODO = {
    'ferramentas.html': 'trabalho',
}
for _a in ['html-puro.html','html-puro-2.html','qualidade.html','movimento.html','motion.html',
           'frameworks.html','back-end.html','landing-pages.html','trafego.html','negocio.html',
           'fluxo.html','ia.html','estudar.html']:
    MODO[_a] = 'estudo'

PAGINA = {
    'estudar.html':        (None, None),
    'html-puro.html':      ('html-puro.html', 'HTML puro'),
    'html-puro-2.html':    ('html-puro.html', 'HTML puro · parte 2'),
    'qualidade.html':      ('qualidade.html', 'Qualidade'),
    'movimento.html':      ('movimento.html', 'Movimento'),
    'motion.html':         ('motion.html', 'Referência: Motion'),
    'frameworks.html':     ('frameworks.html', 'Frameworks'),
    'back-end.html':       ('back-end.html', 'Back-end'),
    'landing-pages.html':  ('landing-pages.html', 'Landing pages'),
    'trafego.html':        ('trafego.html', 'Tráfego'),
    'negocio.html':        ('negocio.html', 'Negócio'),
    'fluxo.html':          ('fluxo.html', 'Fluxo de trabalho'),
    'ia.html':             ('ia.html', 'Trabalhando com IA'),
    'colofao.html':        ('colofao.html', 'Colofão'),
    'patch-notes.html':    ('patch-notes.html', 'Patch notes'),
    'ferramentas.html':    ('ferramentas.html', None),
    'index.html':          None,
    '404.html':            None,
}

def montar_nav(arquivo):
    atual = PAGINA[arquivo]
    href_atual, aqui_texto = atual if atual else (None, None)

    partes = ['<nav class="navmenu" aria-label="Navegação do site">',
              '<details class="navmenu__caixa" id="navmenu">',
              '<summary class="navmenu__botao">Menu<span aria-hidden="true">&#9662;</span></summary>',
              '<div class="navmenu__painel">']
    for rotulo, links, classe_extra in GRUPOS:
        cls = 'navmenu__grupo' + (' ' + classe_extra if classe_extra else '')
        partes.append('<div class="%s">' % cls)
        partes.append('<p class="navmenu__rotulo">%s</p>' % rotulo)
        for href, forte, desc in links:
            atual_attr = ' class="is-atual"' if href == href_atual else ''
            partes.append('<a href="%s"%s><strong>%s</strong><span>%s</span></a>' % (href, atual_attr, forte, desc))
        partes.append('</div>')
    partes.append('</div>')
    partes.append('</details>')
    modo = MODO.get(arquivo)
    partes.append('<div class="modos">')
    partes.append('<a class="modo%s" href="estudar.html">Estudar</a>' %
                  (' is-atual' if modo == 'estudo' else ''))
    partes.append('<a class="modo modo--trabalho%s" href="ferramentas.html">Trabalhar</a>' %
                  (' is-atual' if modo == 'trabalho' else ''))
    partes.append('</div>')
    if aqui_texto:
        partes.append('<span class="navmenu__aqui">%s</span>' % aqui_texto)
    partes.append('</nav>')
    return '\n'.join(partes)

alterados, iguais, problemas = [], [], []
for arq in PAGINA:
    if not os.path.exists(arq):
        print('AVISO: nao existe ->', arq)
        continue
    h = open(arq, encoding='utf-8').read()
    if not re.search(r'<nav class="navmenu" aria-label="[^"]*">.*?</nav>', h, re.S):
        print('ERRO: nao achei o <nav class="navmenu"> em', arq)
        problemas.append(arq)
        continue
    novo_bloco = montar_nav(arq)
    h2 = re.sub(r'<nav class="navmenu" aria-label="[^"]*">.*?</nav>', novo_bloco, h, count=1, flags=re.S)
    if h2 == h:
        iguais.append(arq)
        continue
    open(arq, 'w', encoding='utf-8').write(h2)
    alterados.append(arq)

print('atualizados:', len(alterados), '| ja estavam certos:', len(iguais), '| com problema:', len(problemas))
for a in alterados: print('  atualizado:', a)
for a in problemas: print('  PROBLEMA:', a)

File: scripts/navmenu.mjs
/* ============================================================
   navmenu.mjs — reescreve o <nav class="navmenu"> nas páginas de
   uma vez só. Fonte da verdade do menu: para mudar categoria, link
   ou rótulo, edite GRUPOS aqui e rode — nunca edite o <nav> das
   páginas diretamente, a próxima execução sobrescreve.
   Porta de navmenu.py (Python não está disponível nesta máquina).

   uso:  node scripts/navmenu.mjs
   ============================================================ */
import { existsSync } from 'node:fs';
import { dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';
import { lerTexto, escreverTexto } from './_texto.mjs';
import { FERRAMENTAS } from './_ferramentas.mjs';

// roda a partir da raiz do repositório, não importa de onde foi chamado
process.chdir(join(dirname(fileURLToPath(import.meta.url)), '..'));

// (href, label_forte, label_descricao)
const GRUPOS = [
  ['Construção', [
    ['html-puro.html', 'HTML puro', 'Do zero ao site publicado'],
    ['qualidade.html', 'Qualidade', 'Imagem, a11y, Vitals e testes'],
    ['movimento.html', 'Movimento', 'Animação com critério'],
    ['motion.html', 'Referência: Motion', 'A API inteira, em português'],
    ['frameworks.html', 'Frameworks', 'Formatos além do vanilla'],
    ['back-end.html', 'Back-end', 'Servidor, banco e login'],
  ], null],
  ['Captação e Vendas', [
    ['landing-pages.html', 'Landing pages', 'A estratégia da página'],
    ['trafego.html', 'Tráfego', 'Ser encontrado no Google'],
    ['negocio.html', 'Negócio', 'Preço, escopo e contrato'],
  ], null],
  ['Método', [
    ['fluxo.html', 'Fluxo de trabalho', 'Terminal, Git e DevTools'],
    ['ia.html', 'Trabalhando com IA', 'Brief, referência e revisão'],
    ['vibecoding.html', 'Vibecoding', 'Ferramentas, referências e riscos'],
    ['colofao.html', 'Colofão', 'Como este site foi feito'],
  ], null],
  ['Ferramentas', [
    ['ferramentas.html', 'Ferramentas', 'As quatorze do dia a dia'],
    ['modelos.html', 'Modelos para copiar', 'Efeitos vivos, prontos'],
    ['ferramenta-checklist.html', 'Checklist de entrega', 'Antes de publicar'],
    ['ferramenta-proposta.html', 'Proposta que fecha', 'Os oito pontos'],
    ['ferramenta-briefing.html', 'Briefing', 'As dez perguntas'],
    ['ferramenta-whatsapp.html', 'Link de WhatsApp', 'Com a mensagem pronta'],
    ['ferramenta-utm.html', 'Gerador de UTM', 'Rastrear a origem'],
    ['ferramenta-gerador.html', 'Gerador de comando', 'Onze tipos de pedido'],
  ], 'navmenu__grupo--trabalhar'],
  ['Patch notes', [
    ['patch-notes.html', 'Patch notes', 'O que mudou no site'],
  ], null],
];

// arquivo -> modo do alternador ('estudo', 'trabalho' ou undefined)
const MODO = { 'ferramentas.html': 'trabalho', 'modelos.html': 'trabalho' };
for (const a of [
  'html-puro.html', 'html-puro-2.html', 'qualidade.html', 'movimento.html', 'motion.html',
  'frameworks.html', 'back-end.html', 'landing-pages.html', 'trafego.html', 'negocio.html',
  'fluxo.html', 'ia.html', 'vibecoding.html', 'estudar.html',
]) MODO[a] = 'estudo';
// cada ferramenta tem página própria, na área de trabalho
for (const f of FERRAMENTAS) MODO[f.arquivo] = 'trabalho';

// páginas sem cabeçalho: o hub navega pelo hero e pelos cards. Continua no
// PAGINA abaixo porque o script ainda cuida do <body> e do theme-color dele.
const SEM_MENU = new Set(['index.html']);

// arquivo -> [href que fica 'is-atual', texto do navmenu__aqui] — null = nenhum dos dois (home/404)
const PAGINA = {
  'estudar.html': null,
  'html-puro.html': ['html-puro.html', 'HTML puro'],
  'html-puro-2.html': ['html-puro.html', 'HTML puro · parte 2'],
  'qualidade.html': ['qualidade.html', 'Qualidade'],
  'movimento.html': ['movimento.html', 'Movimento'],
  'motion.html': ['motion.html', 'Referência: Motion'],
  'frameworks.html': ['frameworks.html', 'Frameworks'],
  'back-end.html': ['back-end.html', 'Back-end'],
  'landing-pages.html': ['landing-pages.html', 'Landing pages'],
  'trafego.html': ['trafego.html', 'Tráfego'],
  'negocio.html': ['negocio.html', 'Negócio'],
  'fluxo.html': ['fluxo.html', 'Fluxo de trabalho'],
  'ia.html': ['ia.html', 'Trabalhando com IA'],
  'vibecoding.html': ['vibecoding.html', 'Vibecoding'],
  'colofao.html': ['colofao.html', 'Colofão'],
  'patch-notes.html': ['patch-notes.html', 'Patch notes'],
  'ferramentas.html': ['ferramentas.html', null],
  'modelos.html': ['modelos.html', null],
  'index.html': null,
  '404.html': null,
};
// subpáginas de ferramenta: o menu marca Ferramentas, e a posição diz qual
for (const f of FERRAMENTAS) PAGINA[f.arquivo] = ['ferramentas.html', 'Ferramentas · ' + f.nome];

function montarNav(arquivo) {
  const atual = PAGINA[arquivo];
  const [hrefAtual, aquiTexto] = atual ? atual : [null, null];

  const partes = [
    '<nav class="navmenu" aria-label="Navegação do site">',
    '<details class="navmenu__caixa" id="navmenu">',
    '<summary class="navmenu__botao">Menu<span aria-hidden="true">&#9662;</span></summary>',
    '<div class="navmenu__painel">',
  ];
  for (const [rotulo, links, classeExtra] of GRUPOS) {
    const cls = 'navmenu__grupo' + (classeExtra ? ' ' + classeExtra : '');
    partes.push(`<div class="${cls}">`);
    partes.push(`<p class="navmenu__rotulo">${rotulo}</p>`);
    for (const [href, forte, desc] of links) {
      const atualAttr = href === hrefAtual ? ' class="is-atual"' : '';
      partes.push(`<a href="${href}"${atualAttr}><strong>${forte}</strong><span>${desc}</span></a>`);
    }
    partes.push('</div>');
  }
  partes.push('</div>');
  partes.push('</details>');
  const modo = MODO[arquivo];
  partes.push('<div class="modos">');
  partes.push(`<a class="modo${modo === 'estudo' ? ' is-atual' : ''}" href="estudar.html">Estudar</a>`);
  partes.push(
    `<a class="modo modo--trabalho${modo === 'trabalho' ? ' is-atual' : ''}" href="ferramentas.html">Trabalhar</a>`
  );
  partes.push('</div>');
  // sem rótulo de posição: o cabeçalho é o mesmo em toda página (o texto segue no PAGINA, sem uso no topo)
  partes.push('</nav>');
  return partes.join('\n');
}

// o modo da área no <body> alimenta a troca de --accent por área no CSS.
// Remove antes de adicionar: roda idempotente e limpa página tirada do MODO.
function aplicarModo(html, arquivo) {
  const re = /<body([^>]*)>/;
  const m = html.match(re);
  if (!m) return html;
  const modo = MODO[arquivo];
  let attrs = m[1].replace(/\s*data-modo="[^"]*"/, '');
  if (modo) attrs += ` data-modo="${modo}"`;
  return html.replace(re, () => `<body${attrs}>`);
}

// a cor da barra do navegador acompanha a área (só o tema claro muda;
// no escuro o valor é --ground, igual em todas)
const TEMA_CLARO = { estudo: '#7A4A06', trabalho: '#A8261C' };
function aplicarThemeColor(html, arquivo) {
  const cor = TEMA_CLARO[MODO[arquivo]] || '#0D6B62';
  return html.replace(
    /<meta name="theme-color" content="#[0-9A-Fa-f]{6}" media="\(prefers-color-scheme: light\)">/,
    () => `<meta name="theme-color" content="${cor}" media="(prefers-color-scheme: light)">`
  );
}

const alterados = [];
const iguais = [];
const problemas = [];
for (const arq of Object.keys(PAGINA)) {
  if (!existsSync(arq)) {
    console.log('AVISO: nao existe ->', arq);
    continue;
  }
  const h = lerTexto(arq);
  let h2 = aplicarModo(h, arq);
  h2 = aplicarThemeColor(h2, arq);

  if (!SEM_MENU.has(arq)) {
    if (!/<nav class="navmenu" aria-label="[^"]*">.*?<\/nav>/s.test(h2)) {
      console.log('ERRO: nao achei o <nav class="navmenu"> em', arq);
      problemas.push(arq);
      continue;
    }
    const novoBloco = montarNav(arq);
    h2 = h2.replace(/<nav class="navmenu" aria-label="[^"]*">.*?<\/nav>/s, () => novoBloco);
  }

  if (h2 === h) {
    iguais.push(arq);
    continue;
  }
  escreverTexto(arq, h2);
  alterados.push(arq);
}

console.log('atualizados:', alterados.length, '| ja estavam certos:', iguais.length, '| com problema:', problemas.length);
for (const a of alterados) console.log('  atualizado:', a);
for (const a of problemas) console.log('  PROBLEMA:', a);
